mmv1/products/firebasedatabase/api.yaml

```diff
@@ -0,0 +1,91 @@
+# Copyright 2018 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: FirebaseDatabase
+display_name: Firebase Realtime Database
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://firebasedatabase.googleapis.com/v1beta/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Firebase Realtime Database API
+    url: https://console.cloud.google.com/apis/library/firebasedatabase.googleapis.com/
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Instance'
+    min_version: beta
+    base_url: projects/{{project}}/locations/{{region}}/instances/
+    self_link: projects/{{project}}/locations/{{region}}/instances/{{instance_id}}
+    create_url: projects/{{project}}/locations/{{region}}/instances?databaseId={{instance_id}}
+    description: A Firebase Realtime Database instance.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://firebase.google.com/products/realtime-database'
+      api: 'https://firebase.google.com/docs/reference/rest/database/database-management/rest'
```

Comment:
This documentation part should go to the outside (right before objects)

Reply:
Oh, I see. Didn't realize that it can be at top-level.

Moved

Reply:
Reverted back. This didn't work
```
bundler: failed to load command: compiler (compiler)
RuntimeError: Extraneous variable 'references' in FirebaseDatabase<Api::Product>
  /Users/fredzqm/dev/magic-modules/mmv1/google/yaml_validator.rb:130:in `block in check_extraneous_properties'
  /Users/fredzqm/dev/magic-modules/mmv1/google/yaml_validator.rb:125:in `each'
  /Users/fredzqm/dev/magic-modules/mmv1/google/yaml_validator.rb:125:in `check_extraneous_properties'
  /Users/fredzqm/dev/magic-modules/mmv1/google/yaml_validator.rb:37:in `validate'
  /Users/fredzqm/dev/magic-modules/mmv1/api/object.rb:79:in `validate'
  /Users/fredzqm/dev/magic-modules/mmv1/api/product.rb:62:in `validate'
  compiler:177:in `block in <top (required)>'
  compiler:143:in `each'
  compiler:143:in `<top (required)>'
make[1]: *** [mmv1] Error 1
make: *** [terraform] Error 2
```

Reply:
OK never mind, I somehow always thought it should be outside.